job_creator.php

```diff
@@ -0,0 +1,270 @@
+            $a = array_keys(INSTALLER_TO_PHP_VERSIONS);
+            $a = array_map(fn($k) => (int) $k, $a);
+            sort($a);
+            return $a[count($a) - 1] . 'x-dev';
+        }
+    }
+    
+    public function createJob($phpNum, $opts)
+    {
+        $v = str_replace('.x-dev', '', $this->installerVersion);
+        $v = $v ?: '4';
+        $phpVersions = INSTALLER_TO_PHP_VERSIONS[$v];
+        $default = [
+            # ensure there's a default value for all possible return keys
+            # this allows use to use `if [ "${{ matrix.key }}" == "true" ]; then` in github-actions-ci-cd/ci.yml
+            'installer_version' => $this->installerVersion,
+            'php' => $phpVersions[$phpNum] ?? $phpVersions[count($phpVersions) - 1],
+            'db' => DB_MYSQL_57,
+            'composer_require_extra' => '',
+            'composer_args' => '',
+            'name_suffix' => '',
+            'phpunit' => false,
+            'phpunit_suite' => 'all',
+            'phplinting' => false,
+            'phpcoverage' => false,
+            'endtoend' => false,
+            'endtoend_suite' => 'root',
+            'endtoend_config' => '',
+            'js' => false,
+        ];
+        return array_merge($default, $opts);
+    }
+    
+    private function parseBoolValue($value)
+    {
+        return ($value === true || $value === 'true');
+    }
+
+    private $installerVersion = null;
+    
+    public function createJson($inputs)
+    {
+        // $myRef will either be a branch for push (i.e cron) and pull-request (target branch), or a semver tag
+        $myRef = $inputs['github_my_ref'];
+        $isTag = preg_match('#^[0-9]+\.[0-9]+\.[0-9]+$#', $myRef, $m);
+        $branch = $isTag ? sprintf('%d.%d', $m[1], $m[2]) : $myRef;
+
+        $githubRepository = $inputs['github_repository'];
+        $this->installerVersion = $this->getInstallerVersion($githubRepository, $branch);
+
+        $run = [];
+        $extraJobs = [];
+        $dynamicMatrix = true;
+        $simpleMatrix = false;
+        foreach ($inputs as $input => $value) {
+            if (in_array($input, ['endtoend', 'js', 'phpunit', 'phpcoverage', 'phplinting'])) {
+                $run[$input] = $this->parseBoolValue($value);
+            } else if ($input === 'extra_jobs') {
+                if ($value === 'none') {
+                    $value = [];
+                }
+                $extraJobs = $value;
+                foreach ($extraJobs as $job) {
+                    $job = $this->createJob(3, $job);
+                }
+            } else if ($input === 'dynamic_matrix') {
+                $dynamicMatrix = $this->parseBoolValue($value);
+            } else if ($input === 'simple_matrix') {
+                $simpleMatrix = $this->parseBoolValue($value);
+            }
+        }
+        $matrix = ['include' => []];
+
+        if ($dynamicMatrix) {
+            if ((file_exists('phpunit.xml') || file_exists('phpunit.xml.dist')) && $run['phpunit']) {
```

Comment:
```suggestion
            if ($run['phpunit'] && (file_exists('phpunit.xml') || file_exists('phpunit.xml.dist'))) {
```
No sense checking the filesystem if we don't have to

Reply:
Updated